### PR TITLE
input-remapper: 1.5.0 -> 2.0.1

### PR DESCRIPTION
**Pull request by LunNova**
Corrected data path replacement ~~and moved to python 3.10 as pydbus
doesn't support 3.11~~

https://github.com/sezanzeb/input-remapper/releases/tag/2.0.0
https://github.com/sezanzeb/input-remapper/releases/tag/2.0.1

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [x] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/tools/inputmethods/input-remapper/default.nix [hunk 1]
```diff
@@ -35,13 +35,13 @@ let
...
 (buildPythonApplication {
   pname = "input-remapper";
-  version = "1.5.0";
+  version = "2.0.0";
 
   src = fetchFromGitHub {
-    rev = "e31a1b2bc5d23fe13130afcc242063196335399f";
+    rev = "9e04df79ca4dca77516a9864d740b9b2d4aef788";
     owner = "sezanzeb";
     repo = "input-remapper";
```
Comment (SuperSandro2000) [inline pkgs/tools/inputmethods/input-remapper/default.nix:43]:
```suggestion
    owner = "sezanzeb";
    repo = "input-remapper";
    rev = version;
```

Comment (LunNova) [reply pkgs/tools/inputmethods/input-remapper/default.nix:43]: Not great for the use of final.src.rev on 155

Comment (SuperSandro2000) [reply pkgs/tools/inputmethods/input-remapper/default.nix:43]: IIRC if there are tags those should be used. We could add the commit into an extra variable or leave it out and replace it with a fake hash.

Comment (LunNova) [reply pkgs/tools/inputmethods/input-remapper/default.nix]: Using the rev now

**Comment by LunNova** [comment]
Result of `nixpkgs-review pr 249777` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>input-remapper</li>
    <li>input-remapper.dist</li>
  </ul>
</details>

**Comment by LunNova** [comment]
The nixos test's failing at the moment, need to debug

**Comment by LunNova** [comment]
@ofborg build input-remapper.tests input-remapper

**Comment by Slabity** [comment]
I've been using this fix for my system for about a week without any major issues.

Anything that I can do/test to help this get merged?

**Comment by LunNova** [comment]
I'm happy with it, just needs approved and merged!

File: pkgs/top-level/all-packages.nix [hunk 2]
```diff
@@ -9428,7 +9428,8 @@ with pkgs;
 
   innoextract = callPackage ../tools/archivers/innoextract { };
 
-  input-remapper = python3Packages.callPackage ../tools/inputmethods/input-remapper { };
+  #python310 as pydbus does not build on 3.11
+  input-remapper = python310Packages.callPackage ../tools/inputmethods/input-remapper { };
```
Comment (robert-manchester) [inline pkgs/top-level/all-packages.nix:9432]: pydbus supports python3.11 as of #263669
```suggestion
  input-remapper = python3Packages.callPackage ../tools/inputmethods/input-remapper { };
```

Comment (LunNova) [reply pkgs/top-level/all-packages.nix:9432]: oh nice

**Comment by LunNova** [comment]
@ofborg build input-remapper.tests input-remapper

File: pkgs/top-level/all-packages.nix (hunk 2 above)
Comment (LunNova) [reply pkgs/top-level/all-packages.nix]: reverting that might have made the nixos test hang or this laptop i tested on is just way too slow, asked ofborg to run them

Comment (LunNova) [reply pkgs/top-level/all-packages.nix]: looks like it passed on ofborg

**Review by ghost — APPROVED** [review]
LGTM
Result of `nixpkgs-review pr 249777` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>input-remapper</li>
    <li>input-remapper.dist</li>
  </ul>
</details>

**Comment by LunNova** [comment]
@SuperSandro2000 can you rereview?

File: pkgs/tools/inputmethods/input-remapper/default.nix [hunk 3]
```diff
@@ -33,21 +33,21 @@
...
+(buildPythonApplication rec {
   pname = "input-remapper";
-  version = "1.5.0";
+  version = "2.0.1";
 
   src = fetchFromGitHub {
-    rev = "e31a1b2bc5d23fe13130afcc242063196335399f";
+    rev = version;
     owner = "sezanzeb";
     repo = "input-remapper";
```
Comment (SuperSandro2000) [inline pkgs/tools/inputmethods/input-remapper/default.nix:43]:
```suggestion
    owner = "sezanzeb";
    repo = "input-remapper";
    rev = version;
```

**Review by SuperSandro2000 — APPROVED** [review]

File: pkgs/tools/inputmethods/input-remapper/default.nix (hunk 3 above)
Comment (LunNova) [reply pkgs/tools/inputmethods/input-remapper/default.nix]: Done

**Review by SuperSandro2000 — APPROVED** [review]

File: pkgs/tools/inputmethods/input-remapper/default.nix [hunk 4]
```diff
@@ -33,21 +33,21 @@
...
-    hash = "sha256-KPQLgXSonuOgphagYN2JN+CMIpmjTIPUTCqOPDk0UYU=";
+    rev = version;
+    hash = "sha256-rwlVGF/cWSv6Bsvhrs6nMDQ8avYT80aasrhWyQv55/A=";
   };
 
   postPatch = ''
     # fix FHS paths
     substituteInPlace inputremapper/configs/data.py \
-      --replace "/usr/share/input-remapper"  "$out/usr/share/input-remapper"
+      --replace "/usr/share"  "$out/usr/share"
```
[the remark this answers is not in the document]
Comment (SuperSandro2000) [reply pkgs/tools/inputmethods/input-remapper/default.nix:50]: Good question, the default path in nixpkgs would be just $out/share

[the remark this answers is not in the document]
Comment (LunNova) [reply pkgs/tools/inputmethods/input-remapper/default.nix:50]: The setup.py in the input-remapper repo ends up placing data there. Could change it, means patching more places.

https://github.com/sezanzeb/input-remapper/blob/8f79c25bd966f2ea0fd60c8928df6773911c49f9/setup.py#L115-L130

**Pull request merged**